python/dgl/backend/pytorch/sparse_emb.py

```diff
@@ -0,0 +1,160 @@
+import torch as th
+from .tensor import attach_grad, is_recording
+from .util import get_shared_mem_array, create_shared_mem_array
+from datetime import timedelta
+
+from ... import ndarray as nd
+
+_store = None
+
+class NodeEmbedding: # NodeEmbedding
+    '''Sparse node embeddings for graph.
+
+    DGL provides a sparse embedding to support models that require learnable embeddings.
+    DGL's sparse embeddings are mainly used for learning node embeddings of graph models.
+    The sparse embeddings have to be updated by DGL's optimizers instead of
+    the optimizers provided by the deep learning frameworks (e.g., Pytorch).
+
+    To support efficient training on a graph with many nodes, the embeddings support sparse
+    updates. That is, only the embeddings involved in a mini-batch computation are updated.
+    Currently, DGL provides two optimizer: `SparseAdagrad` and `SparseAdam. DGL will provide more
+    optimizers in the future.
+
+    Parameters
+    ----------
+    num_embeddings : int
+        The number of embeddings. Currently, the number of embeddings has to be the same as
+        the number of nodes or the number of edges.
```

Comment:
does it support heterogeneous graphs? why does the number of embeddings need to be the number of nodes or edges?
since it's called `NodeEmbedding`, should we mention "the number of edges" here?

Reply:
According to the name, we should not mention the number of edges here.